### PR TITLE
[tests] Refactor TorchAO serialization fast tests

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
As discussed internally: https://huggingface.slack.com/archives/C065E480NN9/p1734427523993509?thread_ts=1734425021.147699&cid=C065E480NN9

### CODE REVIEW

File: tests/quantization/torchao/test_torchao.py [hunk 1]
```diff
@@ -506,36 +504,33 @@ def check_serialization_expected_slice(self, expected_slice):
...
-    serialized_expected_slice = expected_slice
-    device = "cpu"
-
-
-class TorchAoSerializationINTA16W8CPUTest(TorchAoSerializationTest):
-    quant_method, quant_method_kwargs = "int8_weight_only", {}
-    expected_slice = np.array([0.3613, -0.127, -0.0223, -0.2539, -0.459, 0.4961, -0.1357, -0.6992, 0.4551])
-    serialized_expected_slice = expected_slice
-    device = "cpu"
+    def test_int_a8w8_cuda(self):
```
Comment (sayakpaul) [inline tests/quantization/torchao/test_torchao.py:507]: Does this need `require_torch_cuda`?

**Review by sayakpaul — COMMENTED** [review]
Maybe we could mark the `_cuda` tests with `require_torch_cuda`?

File: tests/quantization/torchao/test_torchao.py (hunk 1 above)
Comment (a-r-r-o-w) [reply tests/quantization/torchao/test_torchao.py:507]: Oh yes, will mark it one min

Comment (a-r-r-o-w) [reply tests/quantization/torchao/test_torchao.py:507]: Ah, the class that contains these tests already have a require_torch_gpu marker. I think we're good

**Pull request merged**